File: algorithms/0215-kth-largest-element-in-an-array/Solution.java
// Kth Largest Element in an Array - size-k min-heap (PriorityQueue).
// Time O(n log k), Space O(k).

import java.util.PriorityQueue;

class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();

        for (int num : nums) {
            minHeap.add(num);

            if (minHeap.size() > k) minHeap.poll();
        }

        return minHeap.peek();
    }
}

File: algorithms/0215-kth-largest-element-in-an-array/Solution.kt
// Kth Largest Element in an Array - size-k min-heap (PriorityQueue).
// Time O(n log k), Space O(k).

import java.util.PriorityQueue

class Solution {
    fun findKthLargest(nums: IntArray, k: Int): Int {
        val minHeap = PriorityQueue<Int>()

        for (num in nums) {
            minHeap.add(num)

            if (minHeap.size > k) minHeap.poll()
        }

        return minHeap.peek()
    }
}
